src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -135,31 +136,43 @@ def _populate_core_stats_and_sim_state(self, area):
             self.flattened_area_core_stats_dict[area.uuid] = {}
         if self.current_market_time_slot_str == "":
             return
-        core_stats_dict = {'bids': [], 'offers': [], 'trades': [], 'market_fee': 0.0}
-        if hasattr(area.current_market, 'offer_history'):
+        core_stats_dict = {"bids": [], "offers": [], "trades": [], "market_fee": 0.0}
+        if hasattr(area.current_market, "offer_history"):
             for offer in area.current_market.offer_history:
-                core_stats_dict['offers'].append(offer.serializable_dict())
-        if hasattr(area.current_market, 'bid_history'):
+                core_stats_dict["offers"].append(offer.serializable_dict())
+        if hasattr(area.current_market, "bid_history"):
             for bid in area.current_market.bid_history:
-                core_stats_dict['bids'].append(bid.serializable_dict())
-        if hasattr(area.current_market, 'trades'):
+                core_stats_dict["bids"].append(bid.serializable_dict())
+        if hasattr(area.current_market, "trades"):
             for trade in area.current_market.trades:
-                core_stats_dict['trades'].append(trade.serializable_dict())
-        if hasattr(area.current_market, 'market_fee'):
-            core_stats_dict['market_fee'] = area.current_market.market_fee
+                core_stats_dict["trades"].append(trade.serializable_dict())
+        if hasattr(area.current_market, "market_fee"):
+            core_stats_dict["market_fee"] = area.current_market.market_fee
+        if (getattr(area.current_market, "const_fee_rate", None) is not None and
+                area.current_market is not None):
+            core_stats_dict["const_fee_rate"] = (area.current_market.const_fee_rate
+                                                 if area.current_market.const_fee_rate is not None
+                                                 else 0.)
+            core_stats_dict["feed_in_tariff"] = FEED_IN_TARIFF
+            if isinstance(GlobalConfig.market_maker_rate, int):
+                core_stats_dict["market_maker_rate"] = GlobalConfig.market_maker_rate
+            elif isinstance(GlobalConfig.market_maker_rate, dict):
+                core_stats_dict["market_maker_rate"] = \
+                    GlobalConfig.market_maker_rate[area.current_market.time_slot]
```

Comment:
There is a function that does exactly that: https://github.com/gridsingularity/d3a/blob/3c75998cb2b9963030bfebbea5ef4cda6cc625c3/src/d3a/d3a_core/util.py#L502
(BTW; you could refactor the input value `next_market` into `market`, thanks!)